core/storage/file/gae_models.py

```diff
@@ -45,34 +45,96 @@ class FileMetadataModel(base_models.VersionedModel):
 
     @classmethod
     def get_new_id(cls, entity_name):
+        """Raise an exception if an id is not created.
+
+        Args:
+            entity_name: str. An entity name that is tied to the created id.
+        """
         raise NotImplementedError
 
     @classmethod
     def get_undeleted(cls):
+        """Retrieve values that have not been deleted in the file metadata
+        model.
+
+        Returns:
+            A list of undeleted items from the query.
+        """
         return cls.get_all().filter(cls.deleted == False).fetch(  # pylint: disable=singleton-comparison
             feconf.DEFAULT_QUERY_LIMIT)
 
     @classmethod
     def _construct_id(cls, exploration_id, filepath):
+        """ Builds the path for the id to be generated.
```

Comment:
In this and other docstrings, remove the space after the triple quotes.

I don't think I understand this docstring, either. What is this method doing? What is the return value (the ID) used for?